src/auth/models/strategy.class.ts

```diff
@@ -256,14 +258,41 @@ export abstract class Strategy extends events.EventEmitter {
 
     public makeAuthorization = (passport: any) => `Bearer ${passport.user.tokenset.accessToken}`
 
-    public setHeaders = (req: Request, res: Response, next: NextFunction): void => {
+    public setHeaders = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
         if (req.session?.passport?.user) {
-            req.headers['user-roles'] = req.session.passport.user.userinfo.roles.join()
-            req.headers.Authorization = this.makeAuthorization(req.session.passport)
+            if (
+                this.options.routeCredential &&
+                this.options.routeCredential.routes &&
+                this.options.routeCredential.routes.includes(req.url)
```

Comment:
Any issue here with query params? Should you use req.path?
e.g. /foo?bar=thud
req.url == /foo?bar
req.path == /foo

Reply:
@andywilkinshmcts  only reason for using req.url and not path is that i dnt need to know the query string values. Our rule is for the specific routes. Thanks